### PR TITLE
Support get schema by names

### PR DESCRIPTION
**Pull request by Linary**
Change-Id: Id188751189ddd651a32604fb364410bcf857745d

### CODE REVIEW

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/schema/EdgeLabelAPI.java [hunk 1]
```diff
@@ -104,11 +105,20 @@ public String update(@Context GraphManager manager,
     @Timed
     @Produces(APPLICATION_JSON_WITH_CHARSET)
     public String list(@Context GraphManager manager,
-                       @PathParam("graph") String graph) {
-        LOG.debug("Graph [{}] get edge labels", graph);
+                       @PathParam("graph") String graph,
+                       @QueryParam("names") List<String> names) {
+        LOG.debug("Graph [{}] get edge labels by names {}", graph, names);
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/schema/EdgeLabelAPI.java]: not "by names" if names is null

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/schema/EdgeLabelAPI.java [hunk 2]
```diff
@@ -104,11 +106,24 @@ public String update(@Context GraphManager manager,
...
+        if (CollectionUtils.isEmpty(names)) {
+            LOG.debug("Graph [{}] list edge labels", graph);
+        } else {
+            LOG.debug("Graph [{}] get edge labels by names {}", graph, names);
+        }
 
         HugeGraph g = graph(manager, graph);
-        List<EdgeLabel> labels = g.schema().getEdgeLabels();
+        List<EdgeLabel> labels;
+        if (CollectionUtils.isEmpty(names)) {
```
Comment (zhoney) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/schema/EdgeLabelAPI.java]:
```
if (getAll) {
```

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/schema/IndexLabelAPI.java [hunk 3]
```diff
@@ -79,11 +81,24 @@ public String create(@Context GraphManager manager,
     @Timed
     @Produces(APPLICATION_JSON_WITH_CHARSET)
     public String list(@Context GraphManager manager,
-                       @PathParam("graph") String graph) {
-        LOG.debug("Graph [{}] get edge labels", graph);
+                       @PathParam("graph") String graph,
+                       @QueryParam("names") List<String> names) {
+        if (CollectionUtils.isEmpty(names)) {
```
Comment (zhoney) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/schema/IndexLabelAPI.java]: ditto

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/schema/PropertyKeyAPI.java [hunk 4]
```diff
@@ -106,11 +108,24 @@ public String update(@Context GraphManager manager,
     @Timed
     @Produces(APPLICATION_JSON_WITH_CHARSET)
     public String list(@Context GraphManager manager,
-                       @PathParam("graph") String graph) {
-        LOG.debug("Graph [{}] get property keys", graph);
+                       @PathParam("graph") String graph,
+                       @QueryParam("names") List<String> names) {
+        if (CollectionUtils.isEmpty(names)) {
```
Comment (zhoney) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/schema/PropertyKeyAPI.java]: ditto

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/schema/VertexLabelAPI.java [hunk 5]
```diff
@@ -107,11 +109,24 @@ public String update(@Context GraphManager manager,
     @Timed
     @Produces(APPLICATION_JSON_WITH_CHARSET)
     public String list(@Context GraphManager manager,
-                       @PathParam("graph") String graph) {
-        LOG.debug("Graph [{}] get vertex labels", graph);
+                       @PathParam("graph") String graph,
+                       @QueryParam("names") List<String> names) {
+        if (CollectionUtils.isEmpty(names)) {
```
Comment (zhoney) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/schema/VertexLabelAPI.java]: ditto

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/schema/EdgeLabelAPI.java [hunk 6]
```diff
@@ -104,11 +106,24 @@ public String update(@Context GraphManager manager,
     @Timed
     @Produces(APPLICATION_JSON_WITH_CHARSET)
     public String list(@Context GraphManager manager,
-                       @PathParam("graph") String graph) {
-        LOG.debug("Graph [{}] get edge labels", graph);
+                       @PathParam("graph") String graph,
+                       @QueryParam("names") List<String> names) {
+        if (CollectionUtils.isEmpty(names)) {
```
[the remark this answers is not in the document]
Comment (javeme) [reply hugegraph-api/src/main/java/com/baidu/hugegraph/api/schema/EdgeLabelAPI.java]: prefer listAll

**Pull request merged**